CTC/src/ctc/controller/CentralTrafficControlController.java

```diff
@@ -193,20 +195,37 @@ private void connectTables() {
     dispatchPassengersColumn.setCellValueFactory(
         new PropertyValueFactory<TrainTracker, String>("passengers"));
 
-    stopColumn.setCellFactory(TextFieldTableCell.<ScheduleRow>forTableColumn());
+    // set dropdown menu for stations
+    stopColumn.setCellFactory(ComboBoxTableCell.forTableColumn(
+        new DefaultStringConverter(), ctc.getStationList()));
+
     stopColumn.setOnEditCommit(
         (TableColumn.CellEditEvent<ScheduleRow, String> t) -> {
-          ((ScheduleRow) t.getTableView().getItems().get(
-              t.getTablePosition().getRow())
-          ).setStop(t.getNewValue());
+            ((ScheduleRow) t.getTableView().getItems().get(
+                t.getTablePosition().getRow())
+            ).setStop(t.getNewValue());
         });
 
     dwellColumn.setCellFactory(TextFieldTableCell.<ScheduleRow>forTableColumn());
     dwellColumn.setOnEditCommit(
         (TableColumn.CellEditEvent<ScheduleRow, String> t) -> {
-          ((ScheduleRow) t.getTableView().getItems().get(
-              t.getTablePosition().getRow())
-          ).setDwell(t.getNewValue());
+
+          String input = t.getNewValue();
+
+          if (checkTimeFormat(input) || input.equals("")) {
+            ((ScheduleRow) t.getTableView().getItems().get(
+                t.getTablePosition().getRow())
+            ).setDwell(input);
+          } else {
+            AlertWindow alert = new AlertWindow();
+
+            alert.setTitle("Error Submitting");
+            alert.setHeader("Please Use Correct Format");
+            alert.setContent("Please enter the time in the following format: "
+                + "XX:XX:XX");
+
+            alert.show();
```

Comment:
Once a user inputs an illegal format the UI should reject it right? Currently the error comes up and if you hit OK the erroneous value is still added to the table. After more use the value shows up in the table but actually isn't added to the object. 
Reproduce with these steps: 
- Select a track
- Input a train name departure time and block
- Input an erroneous value into a Dwell cell.
- Click Ok on the alert message 
- click to add a new dwell time in another row
- see that the erroneous value wasn't deleted
- add the train to the queue 
- click on the queued train and see there is no dwell time.

Reply:
Yeahhh.. I noticed this too. With the way the TableView component works, I'm not sure if I can do anything about this... Thats why I added the error message to alert them. I think it's the best I can do for now.